src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx

```diff
@@ -21,67 +21,85 @@ export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
   animations,
   activeAnimation,
   scale = 1,
+  setModelFallback,
   onLoaded,
   emotion,
   bloom
 }) => {
-  const groupRef = useRef<Group>(null);
   const mixerRef = useRef<AnimationMixer | null>(null);
+  const activeActionRef = useRef<AnimationAction | null>(null);
+  const animationTimeRef = useRef<number>(0);
+
   const [loadedAnimations, setLoadedAnimations] = useState<Record<string, AnimationClip>>({});
-  const [activeAction, setActiveAction] = useState<AnimationAction | null>(null);
 
   const { scene } = useGltfCachedLoader(modelSrc);
   const { nodes } = useGraph(scene);
 
   useEffect(() => {
-    if (scene && groupRef.current) {
-      groupRef.current.add(scene);
+    const loadAllAnimations = async () => {
+      const clips: Record<string, AnimationClip> = {};
+
+      await Promise.all(
+        Object.keys(animations).map(async (name) => {
+          const newClip = await loadAnimationClip(animations[name]);
+          clips[name] = newClip;
+        })
+      );
+
+      setLoadedAnimations(clips);
+    };
 
-      mixerRef.current = new AnimationMixer(scene);
+    loadAllAnimations();
+  }, [animations]);
+
+  useEffect(() => {
+    if (scene) {
+      const mixer = new AnimationMixer(scene);
+      mixerRef.current = mixer;
+
+      if (activeActionRef.current) {
+        const newAction = mixer.clipAction(activeActionRef.current.getClip());
+
+        newAction.play();
+        mixer.update(animationTimeRef.current);
+
+        activeActionRef.current = newAction;
+      }
     }
 
     return () => {
       mixerRef.current?.stopAllAction();
+      mixerRef.current?.uncacheRoot(scene);
     };
   }, [scene]);
 
-  useEffect(() => {
-    const loader = new GLTFLoader();
-    Object.keys(animations).forEach((name) => {
-      loader.load(animations[name], (gltf) => {
-        const newClip = gltf.animations[0];
-        setLoadedAnimations((prev) => ({
-          ...prev,
-          [name]: newClip
-        }));
-      });
-    });
-  }, [animations]);
-
   useEffect(() => {
     const mixer = mixerRef.current;
+    const prevAction = activeActionRef.current;
     const newClip = loadedAnimations[activeAnimation];
 
     if (!newClip || !mixer) return;
+    if (prevAction && prevAction.getClip().name === newClip.name) return;
 
     const newAction = mixer.clipAction(newClip);
-    const prevAction = activeAction;
+    activeActionRef.current = newAction;
 
     if (prevAction) {
       newAction.reset();
       newAction.crossFadeFrom(prevAction, 0.5, true);
```

Comment:
I think it could be nice to make this configurable from a config or prop.

Reply:
Introduced an animationConfig which is used by the MultipleAnimationModel component